relayer/chains/substrate/event_parser.go

```diff
@@ -27,3 +117,209 @@ func (c clientInfo) ClientState() provider.ClientState {
 		ConsensusHeight: c.ConsensusHeight,
 	}
 }
+
+func (res *clientInfo) MarshalLogObject(enc zapcore.ObjectEncoder) error {
+	enc.AddString("client_id", res.ClientID)
+	enc.AddUint64("consensus_height", res.ConsensusHeight.RevisionHeight)
+	enc.AddUint64("consensus_height_revision", res.ConsensusHeight.RevisionNumber)
+	return nil
+}
+
+func (res *clientInfo) parseAttrs(log *zap.Logger, attributes interface{}) {
```

Comment:
is there a reason for which this function does not return an error?

Reply:
Errors are being logged in `parseAttr`, The IBC message is populated from the events, so there is no way to return the error to the source, So in these methods, the errors is being logged. @agouin can clarify it better.

Reply:
In the cosmos event parser, it makes a best effort attempt to parse all of the attributes that are present, but does not quit if there is an issue parsing a single attribute. This is consistent here also.